src/lightning/pytorch/CHANGELOG.md

```diff
@@ -174,6 +174,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 
+- Fixed FSDP full-precision `param_dtype` training (`16-mixed`, `bf16-mixed` and `32-true` configurations) to avoid FSDP assertion errors with PyTorch < 2.0 ([#18278](https://github.com/Lightning-AI/lightning/pull/18278))
+
+
+- Issue warnings rather than failing on optim state dict loading/saving with `FSDPStrategy` PyTorch < 2.0 ([#18278](https://github.com/Lightning-AI/lightning/pull/18278))
```

Comment:
Could we do this in a separate PR? The two issues aren't related afaik, so they should be addressed separately IMO. I also have a branch where I worked on this for #18230

Reply:
Absolutely, makes sense. If you have a planned PR you're working on for #18230, I'm happy to let you continue with that (since I told you I might not have a chance to implement, totally my fault!). Some tests for this PR won't succeed I think without some temp #18230 fix though so, we can just wait for your PR to land first. If instead you want me to submit my #18230 fixes as a separate PR I'm happy to do that too, just lmk either way.

Reply:
Looks great! Thanks for helping with this. I rebased my branch on top of yours and submitted https://github.com/Lightning-AI/lightning/pull/18296, which we can land after your PR here.